### PR TITLE
fix: #3 Setup correct authentication scheme

### PR DESCRIPTION
**Pull request by Jay-Dee**
#3  Setup correct authentication scheme as per https://github.com/Identityerver/IdentityServer4.AccessTokenValidation/issues/72#issuecomment-329054417

Now correctly responds with 401 for unauthorized instead of 404
image

### CODE REVIEW

**Comment by meysamhadeli** [comment]
> #3 Setup correct authentication scheme as per https://github.com/Identityerver/IdentityServer4.AccessTokenValidation/issues/72#issuecomment-329054417
> 
> Now correctly responds with 401 for unauthorized instead of 404 image

Thanks for the solution. It works completely. Please consider this small change.
But I have a suggestion instead of pass `AuthenticationSchemes = IdentityServerAuthenticationDefaults.AuthenticationScheme` value to `Authorize` attribute, You can configure it globally in `AddJwt` method like below and it's more cleaner.

```csharp
using BuildingBlocks.Utils;
using BuildingBlocks.Web;
using Duende.IdentityServer.Models;
using IdentityServer4.AccessTokenValidation;
using Microsoft.AspNetCore.Authentication.JwtBearer;
using Microsoft.Extensions.DependencyInjection;

namespace BuildingBlocks.Jwt;

public static class JwtExtensions
{
    public static IServiceCollection AddJwt(this IServiceCollection services)
    {
        var jwtOptions = services.GetOptions<JwtBearerOptions>("Jwt");

        services.AddAuthentication(options =>
            {
                options.DefaultScheme = IdentityServerAuthenticationDefaults.AuthenticationScheme;
                options.DefaultAuthenticateScheme = IdentityServerAuthenticationDefaults.AuthenticationScheme;
                options.DefaultChallengeScheme = IdentityServerAuthenticationDefaults.AuthenticationScheme;
            })
            .AddJwtBearer(JwtBearerDefaults.AuthenticationScheme, options =>
            {
                options.Authority = jwtOptions.Authority;
                options.TokenValidationParameters.ValidateAudience = false;
            });

        if (!string.IsNullOrEmpty(jwtOptions.Audience))
        {
            services.AddAuthorization(options =>
                options.AddPolicy(nameof(ApiScope), policy =>
                {
                    policy.RequireAuthenticatedUser();
                    policy.RequireClaim("scope", jwtOptions.Audience);
                })
            );
        }

        return services;
    }
}
```

File: src/Modules/Booking/src/Booking/Booking/Features/CreateBooking/CreateBookingEndpoint.cs [hunk 1]
```diff
@@ -10,7 +11,7 @@ namespace Booking.Booking.Features.CreateBooking;
 public class CreateBookingEndpoint : BaseController
 {
     [HttpPost]
-    [Authorize]
+    [Authorize(AuthenticationSchemes = IdentityServerAuthenticationDefaults.AuthenticationScheme)]
```
Comment (meysamhadeli) [inline src/Modules/Booking/src/Booking/Booking/Features/CreateBooking/CreateBookingEndpoint.cs:14]: Please remove `AuthenticationSchemes = IdentityServerAuthenticationDefaults.AuthenticationScheme` from here and configure it globally in `AddJwt` method.

**Comment by Jay-Dee** [comment]
Hi, that makes sense. I have prepared the change but having issues with the testing. Will send a PR when ready

**Comment by meysamhadeli** [comment]
> Hi, that makes sense. I have prepared the change but having issues with the testing. Will send a PR when ready

Ok. Thank You :)

**Comment by meysamhadeli** [comment]
> Hi, that makes sense. I have prepared the change but having issues with the testing. Will send a PR when ready



> Hi, that makes sense. I have prepared the change but having issues with the testing. Will send a PR when ready

Hi, Is the problem to run the project not resolved?

**Pull request closed without merging**